src/backend/cuda/fft.cpp

```diff
@@ -25,11 +27,15 @@ using std::string;
 namespace cuda
 {
 
+typedef std::pair<string, cufftHandle> FFTPlanPair;
+typedef std::deque<FFTPlanPair> FFTPlanCache;
```

Comment:
If we used a map we can get by using only one data type. Considering that the size is going to be pretty small,I don't think it will affect the performance.